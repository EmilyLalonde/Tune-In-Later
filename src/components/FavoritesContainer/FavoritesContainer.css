.FavoritesContainer {
    box-sizing: border-box;
    display: flex;
}