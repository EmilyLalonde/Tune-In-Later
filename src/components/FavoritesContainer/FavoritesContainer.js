import "./FavoritesContainer.css";
import React from "react";
import Album from "../Album/Album";
import { connect } from 'react-redux';

export const FavoritesContainer = ({ favorites, handleFavorite }) => {
  console.log('in FC', handleFavorite)
  const idsOfAllFavs = [];
  if(favorites){  
    favorites.forEach(fav => idsOfAllFavs.push(fav.album_id))
  }
  
  const isThisAFav = (album) =>{
  if(idsOfAllFavs.includes(album.album_id)){
    return true
  }else {
    return false
  }
}
  const favoriteAlbumCards = favorites.map(album => {
    return <Album isFav={isThisAFav(album)} key={album.collectionId} albumData={album} handleFavorite={handleFavorite}/>;
  });

  return (
    <section className="FavoritesContainer">
        {favoriteAlbumCards}
    </section>
    );
};

const mapStateToProps = state => ({
  favorites: state.favorites
})

export default connect(mapStateToProps)(FavoritesContainer);