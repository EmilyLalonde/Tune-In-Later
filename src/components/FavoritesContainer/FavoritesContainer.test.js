import React from 'react';
import { shallow } from 'enzyme'
import FavoritesContainer from './FavoritesContainer';

describe('FavoritesContainer', () => {
  it('should', () => {
    expect(true).toEqual(true)
  })
})