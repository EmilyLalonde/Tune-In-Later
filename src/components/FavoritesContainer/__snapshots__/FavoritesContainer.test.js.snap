// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer should match the snapshot 1`] = `
<section
  className="FavoritesContainer"
>
  <Album
    albumData={Object {}}
    isFav={true}
  />
</section>
`;
